src/javascript/app_2/App/Components/Layout/Footer/__tests__/toggle_fullscreen.spec.js

```diff
@@ -0,0 +1,54 @@
+import React                         from 'react';
+import { expect }                    from 'chai';
+import { spy }                       from 'sinon';
+import { configure, shallow, mount } from 'enzyme';
+import Adapter                       from 'enzyme-adapter-react-16';
+import { ToggleFullScreen }          from '../toggle_fullscreen.jsx';
+import { IconMaximize }              from '../../../../../Assets/Footer';
+
+configure({ adapter: new Adapter() });
+spy(ToggleFullScreen.prototype, 'componentDidMount');
+
+describe('ToggleFullScreen', () => {
+    it('should render one <ToggleFullScreen /> component', () => {
+        const wrapper = shallow(<ToggleFullScreen  />);
+        expect(wrapper).to.have.length(1);
+    });
+    it('should call componentWillMount', () => {
+        mount(<ToggleFullScreen  />);
+        expect(ToggleFullScreen.prototype.componentDidMount).to.have.property('called', true);
+    });
+    it('should have .ic-fullscreen', () => {
+        const wrapper = shallow(<ToggleFullScreen  />);
+        expect(wrapper.find('.ic-fullscreen').exists()).to.be.true;
+    });
+    it('should not have .active if it\'s false in state', () => {
+        const wrapper = shallow(<ToggleFullScreen  />);
+        expect(wrapper.find('.active').exists()).to.be.false;
+    });
+    it('should have .active if it\'s true in state', () => {
+        const wrapper = shallow(<ToggleFullScreen  />);
+        wrapper.setState({ is_full_screen: true });
+        expect(wrapper.find('.active').exists()).to.be.true;
+    });
+    it('should contain <IconMaximize />', () => {
+        const wrapper = shallow(<ToggleFullScreen />);
+        expect(wrapper.contains(<IconMaximize className='footer-icon' />)).to.be.true;
+    });
+    it('should have onclick property as an instance of a Function', () => {
+        const wrapper = shallow(<ToggleFullScreen />);
+        expect(wrapper.prop('onClick')).to.be.an.instanceof(Function);
+    });
+    it('should set is_full_screen equal to false in state, after onclick, when is_full_screen is false and there is no browser', () => {
+        const wrapper = mount(<ToggleFullScreen />);
+        wrapper.setState({ is_full_screen: false });
+        wrapper.find('a').prop('onClick')({stopPropagation: () => {}});
+        expect(wrapper.state().is_full_screen).to.be.false;
+    });
+    it('should set is_full_screen equal to false in state, after onclick, when is_full_screen is true and there is no browser', () => {
+        const wrapper = mount(<ToggleFullScreen />);
+        wrapper.setState({ is_full_screen: true });
+        wrapper.find('a').prop('onClick')({stopPropagation: () => {}});
+        expect(wrapper.state().is_full_screen).to.be.false;
+    });
```

Comment:
I think we should add one or multiple test case(s) that check correct event registration. Also we need to make sure that the correct handler is being called whenever the relative event is triggered. [see here](https://medium.com/@DavideRama/testing-global-event-listener-within-a-react-component-b9d661e59953) 

For example, we need to assert that `fullscreenchange` triggers `onFullScreen` method.

Reply:
could you please explain more? How to listen to document events with enzyme (No jest)?

Reply:
I find no reason to do it without jest, as test runner solves this problem real good.